implementations/elixir/ockam/ockam/lib/ockam/transport/portal/interceptor.ex

```diff
@@ -0,0 +1,274 @@
+defmodule Ockam.Transport.Portal.Interceptor do
+  @moduledoc """
+  Interceptor worker for portals.
+
+  Can be inserted in the route between inlet and outlet.
+  Messages on outer address are coming from the inlet, messages on inner address
+  are coming from the outlet.
+
+  Supports `:init_message` handling from `Ockam.Session.Spawner`
+
+  Supports message reconstruction from multiple tunnel messages (batching).
+
+  Messages are parsed and processed by behaviour functions
+  For payload messages:
+  `handle_outer_payload`
+  `handle_inner_payload`
+
+  For signal messages (:ping, :pong, :disconnect):
+  `handle_outer_signal`
+  `handle_inner_signal`
+
+  :disconnect stops the interceptor worker
+
+  Other messages are forwarded as is.
+
+  `outer` and `inner` depends on which direction message flow created the interceptor,
+  usually it's configured in the route of the inlet, so `outer` is inlet and `inner` is outlet.
+
+  Options:
+  :interceptor_mod - module implementing interceptor behaviour
+  :interceptor_options - options to initialize interceptor with
+  :init_message - message from `Ockam.Session.Spawner`
+  """
+  use Ockam.AsymmetricWorker
+
+  alias Ockam.Message
+  alias Ockam.Router
+  alias Ockam.Transport.Portal.TunnelProtocol
+
+  ## TODO: enforce that on inlet/outlet level
+  @max_payload_size 48 * 1024
```

Comment:
Because of https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_receiver.rs#L9
It's an arbitrary number we chose for our portals.